Source/Plugin.BLE.Android/Adapter.cs

```diff
@@ -149,6 +149,22 @@ private void StartScanningNew(ScanFilterOptions scanFilterOptions)
 
             var ssb = new ScanSettings.Builder();
             ssb.SetScanMode(ScanMode.ToNative());
+
+            if (Build.VERSION.SdkInt >= BuildVersionCodes.M)
+            {
+                // set the match mode on Android 6 and above
+                ssb.SetMatchMode(ScanMatchMode.ToNative());
+
+                // If set to agressive, reduce the number of adverts needed before raising the DeviceFound callback
+                if (ScanMatchMode.ToNative() == BluetoothScanMatchMode.Aggressive)
+                {
+                    // Be more agressive when seeking adverts
+                    ssb.SetNumOfMatches((int)BluetoothScanMatchNumber.OneAdvertisement);
+                    ssb.SetCallbackType(ScanCallbackType.AllMatches);
```

Comment:
@Bodwad  This call to `SetCallbackType` is not really necessary, since `AllMatches` is the default, right?

Reply:
> @Bodwad This call to `SetCallbackType` is not really necessary, since `AllMatches` is the default, right?

I was trying to work this out as I had noticed the commented out line that already existed in Master, I presume this is why it got commented out. 

There doesn't seem to be anything in the Android documentation about default values. As a result I would personally specify it as we know that is definitely what we want in this scenario. It would also override any potential 'Manufacturer OS' defaults that may exist.

If you think it isn't needed then feel free to remove it

Reply:
`AllMatches` indeed seems to be the default, and it should really be used for all scenarios (not just the aggressive mode).

It doesn't matter much if we set it explicitly, but code-wise it looks weird if it's only set for aggressive. So I'd rather not set it at all (will remove that line).